packages/@react-spectrum/calendar/src/CalendarBase.tsx

```diff
@@ -60,24 +61,49 @@ export function CalendarBase<T extends CalendarState | RangeCalendarState>(props
     timeZone: state.timeZone
   });
 
-  let titles = [];
   let calendars = [];
   for (let i = 0; i < visibleMonths; i++) {
-    titles.push(
-      <div key={i} className={classNames(styles, 'spectrum-Calendar-monthHeader')}>
+    let d = currentMonth.add({months: i});
+    calendars.push(
+      <div key={i} className={classNames(styles, 'spectrum-Calendar-month')}>
+        {/* Put the heading first so it is the first thing touch screen reader users encounter. */}
+        <h2
+          // If displaying more than one month, we have a visually hidden heading describing
+          // the entire visible range, and the calendar itself describes the individual month
+          // so we don't need to repeat that here for screen reader users.
+          aria-hidden={visibleMonths > 1 || undefined}
+          className={classNames(styles, 'spectrum-Calendar-title')}>
+          {monthDateFormatter.format(d.toDate(state.timeZone))}
+        </h2>
         {i === 0 &&
+          // Next, put the previous button, so it's easy to adjust the month if not right.
+          // With a keyboard, this will be the first tab stop, but desktop screen readers are
+          // better about reading the role="group" label as well so there is enough context.
           <ActionButton
             {...prevButtonProps}
             UNSAFE_className={classNames(styles, 'spectrum-Calendar-prevMonth')}
             isQuiet>
             {direction === 'rtl' ? <ChevronRight /> : <ChevronLeft />}
           </ActionButton>
         }
-        <h2
-          className={classNames(styles, 'spectrum-Calendar-title')}>
-          {monthDateFormatter.format(currentMonth.add({months: i}).toDate(state.timeZone))}
-        </h2>
+        {i === 0 &&
+          // And for touch screen readers, add a visually hidden next button as well since it
+          // would be tedious to get past the entire grid of days to get to the visual next button.
+          <VisuallyHidden>
+            <button
+              aria-label={nextButtonProps['aria-label']}
+              disabled={nextButtonProps.isDisabled}
+              onClick={() => state.focusNextPage()}
+              tabIndex={-1} />
+          </VisuallyHidden>
```

Comment:
I wonder if it might make more sense if the VisuallyHidden Next button were after the Calendar grid and the visible next button were to remain before the grid.

Reply:
Personally I found it to make sense that if you're on the date grid, you can shift tab to get to the previous button and tab to get to the next button. If we put them both before, then you have to tab through both to get to the dates, and shift tab to get to both buttons from the dates. What made you think putting the next button first would make more sense?

Reply:
The previous and next month buttons in native `input[type=date]` control are both before the table. Same is true for https://www.w3.org/TR/wai-aria-practices/examples/dialog-modal/datepicker-dialog.html. Feels more intuitive tabbing left to right, top to bottom.